qiskit/circuit/library/phase_oracle.py

```diff
@@ -63,6 +63,11 @@ def __init__(
             var_order(list): A list with the order in which variables will be created.
                (default: by appearance)
         """
+        # ignore deprecation warning for BooleanExpression; implementation is changing in 2.0
+        import warnings
+
+        warnings.simplefilter("ignore", DeprecationWarning)
```

Comment:
This will enable a global filter, we should use 
```python
with warnings.catch_warnings(action="ignore", category=DeprecationWarning):
   # code here...
```
🙂

Reply:
Done in b4878c62c